New tutorial (#57)
trixi-framework/HOHQMesh.jl: docs/src/tutorials/symmetric_mesh.md

                                      [0.75, 1.75, 0.0])

line7 = newEndPointsLineCurve("bump", [0.75, 1.75, 0.0],
                                      [1.0, 1.75, 0.0])

line8 = newEndPointsLineCurve("right", [1.0, 1.75, 0.0],
                                       [1.0, 2.0, 0.0])

line9 = newEndPointsLineCurve("top", [1.0, 2.0, 0.0],
                                     [-0.05, 2.0, 0.0])
```
The given boundary names will also be the element boundary names written to the mesh file.
The only exception is the first boundary curve that is given
the name `"symmetry"`. The name of this outer boundary curve is a special keyword
in HOHQMesh that says it is the straight line across which
a reflection will occur.

!!! tip "Name of the symmetry boundary"
    As noted above, `"symmetry"` is a keyword for the HOHQMesh generator that
    prescribes over which line a reflection of the mesh occurs. The Fortran implementation
    for this keyword in HOHQMesh is not case sensitive. So, within the
    Julia script one can name this reflection boundary line `"symmetry"` or `"Symmetry"`
    or `"SYMMETRY"` or even something strange like `"sYmMeTrY"` and the mesh will still

>     generate successfully.

Reviewer: Does it have to be such a widely and, imho, accidentally used name? Wouldn't it be possible to pick a name that would be invalid for a boundary name (e.g., "{symmetry}" or ":symmetry" etc.) such that no accidental use can happen? Or, if you want to keep it a valid boundary name for implementation reasons, make it something "unusual" such as "__symmetry__".

If "symmetry" it shall remain, I suggest to prominently document this in the main docs/tutorials to highlight that this is a special name.

Author: David and I discussed this name a bit and settled on "symmetry". However, making a specialized variant of the name to avoid accidental misuse is a good idea.

> !!! tip "Name of the symmetry boundary"
>     As noted above, `"symmetry"` is a keyword for the HOHQMesh generator that
>     prescribes over which line a reflection of the mesh occurs. The Fortran implementation
>     for this keyword in HOHQMesh is not case sensitive. So, within the
>     Julia script one can name this reflection boundary line `"symmetry"` or `"Symmetry"`
>     or `"SYMMETRY"` or even something strange like `"sYmMeTrY"` and the mesh will still
>     generate successfully.

Reviewer: As a second note, I'd recommend to not confuse users with implementation details. I'd just enforce the lower-case version in Julia and that's it 🤷‍♂️

If this is too much of a violation of your sense for consistency (which I would totally understand, with me being me 😬), just shorten it to something like
```suggestion
!!! tip "Name of the symmetry boundary is case-insensitive"
    The name of the reflection boundary line is not case-sensitive, thus`"symmetry"` or `"Symmetry"` or `"SYMMETRY"` are all valid and will be recognized as the keyword for symmetry.
```

Author: Sure, we can avoid getting users bogged down in such details.

    in their construction above. The boundary labeled `"symmetry"` is now internal and
    is marked appropriately as such. The reflected boundary names are appended
    with `_R` (for reflected) in the mesh file. For instance, the reflected version
    of the boundary `bottom` has the name `bottom_R` or the boundary named `circle` has the
    reflected boundary counterpart named `circle_R`. These can be changed as desired by editing the mesh file.

## Changing the reflection line

It is also possible to create a symmetry boundary composed of multiple be co-linear segments.

To change the line along which the mesh is reflected, we remove the current mesh
that was just generated and re-plot the model curves
and background grid.
```julia
remove_mesh!(symmetric_mesh)
updatePlot!(symmetric_mesh, MODEL+GRID)
```
The `remove_mesh!` command deletes the mesh information from
the interactive mesh project `symmetric_mesh` as well as the mesh file `symmetric_mesh.mesh`,
control file `symmetric_mesh.control`, plot file `symmetric_mesh.tec`,
and mesh statistics file `symmetric_mesh.txt` from the `out` folder.

> To illustrate the reflection about multiple boundary curves (which must be co-linear!),

Reviewer: Is there a check for co-linearity? If not, it might be good to note that this is something that is not enforced and that the user is responsible for correct use.

If there is a check (which would make sense imho), it would be nice to inform the user about it as well here.

Reviewer 2: Yes, an exception is thrown and the user is informed that the reflection was not performed. The user is also informed if the curve along which the reflection is requested is not a straight line. The tests for the latter are not exhaustive. It basically checks to see if the second derivative (that code already existed) is zero at a couple of points.

At the end of this tutorial one can find the scripts necessary to generate the meshes
described herein.

### Synopsis

This tutorial demonstrates how to:
* Indicate a symmetry boundary line.
* Construct an outer boundary with several connected curves.
* Add the background grid when an outer boundary curve is present.
* Visualize an interactive mesh project.

## Initialization

From a Julia REPL we load the HOHQMesh package as well as
[GLMakie](https://github.com/JuliaPlots/GLMakie.jl/), a backend of
[Makie.jl](https://github.com/JuliaPlots/Makie.jl/), to visualize the
curves, mesh, etc. from the interactive tool.
```julia
julia> using GLMakie, HOHQMesh
```
Now we are ready to interactively generate unstructured quadrilateral meshes!

We create a new project with the name `"symmetric_mesh"` and
assign `"out"` to be the folder where any output files from the mesh generation process
will be saved. By default, the output files created by HOHQMesh will carry the same name
as the project. For example, the resulting HOHQMesh control file from this tutorial
will be named `symmetric_mesh.control`.
If the folder `out` does not exist, it will be created automatically in

> the current file path.

Reviewer: ```suggestion
the current working directory.
```
Clearer to me, but YMMV 🤷‍♂️
